### ISSUE TITLE
When  metrics 'reset_state' during training. 

### ISSUE THREAD

**Issue by Jhangsy** [keras-team/keras#14594]
I noticed that Keras removed metrics like F1 since it is a global metrics and metrics is an 'average over batch'. 
However, when I do some experiments using my own ```sparse_categorical_true_positives``` metric function, I found it both 
```sparse_categorical_accuracy```(built-in metrics) and ```sparse_categorical_true_positives``` are cumulated cross batches.
As in the picture,  ```sparse_categorical_accuracy``` =  ```sparse_categorical_true_positives```/93854(total samples) ， it looks like both metrics are global not batch. 
<img width="1348" alt="Screen Shot 2021-04-14 at 5 52 38 PM" src="https://user-images.githubusercontent.com/16949651/114691409-3b1f9b80-9d4a-11eb-9a6d-55603876b163.png">
```
class SparseCategoricalTruePositives(tf.keras.metrics.Metric):

    def __init__(self, name='sparse_categorical_true_positives', **kwargs):
      super(SparseCategoricalTruePositives, self).__init__(name=name, **kwargs)
      self.true_positives = self.add_weight(name='tp', initializer='zeros')

    def update_state(self, y_true, y_pred, sample_weight=None):

      y_true = K.reshape(K.cast(y_true, tf.int64), [-1])
      y_pred = K.reshape(K.argmax(y_pred, axis=-1), [-1])

      values = K.sum(K.cast(K.equal(y_true, y_pred), self.dtype))
      if sample_weight is not None:
        sample_weight = K.cast(sample_weight, self.dtype)
        values = K.multiply(values, sample_weight)
      self.true_positives.assign_add(values)

    def result(self):
      return self.true_positives
```

**Comment by sushreebarsa** [comment]
@Jhangsy  In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!
